draft-sharabayko-mops-srt.md

```diff
@@ -2086,15 +2086,583 @@ and RTTVar are updated by the socket based on algorithms for the sender
 
-SRT provides certain mechanisms for the sender to get some feedback from the receiving side through the ACK packets ({{ctrl-pkt-ack}}). Every 10 ms the sender receives the latest values of RTT and RTT variance, Available Buffer Size, Packets Receiving Rate and Estimated Link Capacity. Upon reception of the NAK packet ({{ctrl-pkt-nak}}) the sender can detect packet losses during the transmission. These mechanisms provide a solid background for various congestion control algorithms.
+   ~~~
+   MAX_BW = INPUT_BW * (1 + OVERHEAD /100)
+   ~~~
+
+   Note that INPUTBW mode reduces to the MAXBW mode.
+
+3. Measuring SRT sender's input rate internally and setting the
+   overhead OVERHEAD (ESTINPUTBW mode).
+
+   In this mode, SRT adjusts the value of maximum bandwidth each time it
+   gets the updated estimation of input rate EST_INPUT_BW:
+
+   ~~~
+   MAX_BW = EST_INPUT_BW * (1 + OVERHEAD /100)
+   ~~~
+
+4. A hybrid mode which takes the configured input rate INPUT_BW,
+   compares it against estimated value EST_INPUT_BW, and uses the
+   maximum of two (HYBRIDBW mode):
+
+   ~~~
+   MAX_BW = max(INPUT_BW, EST_INPUT_BW) * (1 + OVERHEAD /100)
+   ~~~
+
+Note that the units of MAX_BW, INPUT_BW, and EST_INPUT_BW are bytes per
+second. OVERHEAD is defined in %.
+
+Both ESTINPUTBW and HYBRIDBW modes are recommended for setting the
+maximum bandwidth as it (MAX_BW) follows the fluctuations in
+SRT sender's input rate. However, there are certain considerations
+that should be taken into account.
+
+In INPUTBW mode, SRT takes as an input the rate that had been
+configured as the expected output rate of an encoder (in terms
+of bitrate for the packets including audio and overhead). But it is
+normal for an encoder to occasionally overshoot. At low bitrate,
+sometimes an encoder can be too optimistic and will output more bits
+than expected. Under these conditions, SRT packets would not go out
+fast enough due to too low value of the configured bandwidth limitation.
+
+This is mitigated by calculating the bitrate internally (ESTINPUTBW
+mode). SRT examines the packets being submitted and calculates the
+input rate as a moving average. However, this introduces a bit of a
+delay based on the content. It also means that if an encoder encounters
+black screens or still frames, this would dramatically lower the
+bitrate being measured, which would in turn reduce the SRT output rate.
+And then, when the video picks up again, the input rate rises sharply.
+SRT would not start up again fast enough on output because of the time
+it takes to measure the speed. Packets might be accumulated in the
+SRT's sender buffer and, as a result, delayed causing their too late
+arrival at the decoder and possible drops by the receiver.
+
+A proposed solution is to have SRT take the input rate configuration
+from the encoder, measure the actual input, and use whichever value
+is larger (HYBRIDBW mode). Combining the two methods overcomes
+the deficiencies of each.
```

Comment:
```suggestion
HYBRIDBW mode combines the two methods above and overcomes 
the deficiencies of each. SRT takes the input rate configuration from the 
encoder, measures the actual input, and uses whichever value is larger.
```

Reply:
I also agree with this suggestion, as using the term "proposed solution" is too loud for such case. Each mode has its use case. Let us not put any of them in favor.